addons/mrp_repair/tests/test_flows.py

```diff
@@ -0,0 +1,46 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import Form, tagged
+from odoo.addons.mrp.tests.common import TestMrpCommon
+
+
+@tagged('post_install', '-at_install')
+class TestMrpRepairFlows(TestMrpCommon):
+    def test_possible_to_add_kit_after_confirm(self):
+        """
+        Test that it is possible to add a kit manufactured product to an already confirmed Repair Order
+        """
+        repaired, part1, part2, kit = self.env['product.product'].create([
+            {'name': 'Repaired'},
+            {'name': 'Kit Component1'},
+            {'name': 'Kit Component2'},
+            {'name': 'Kit', 'type': 'product'},
+        ])
+        CREATE = 0
+        self.env['mrp.bom'].create({
+            'product_id': kit.id,
+            'product_tmpl_id': kit.product_tmpl_id.id,
+            'type': 'phantom',
+            'bom_line_ids': [
+                (CREATE, 0, {
+                    'product_id': part1.id,
+                    'product_tmpl_id': part1.product_tmpl_id.id,
+                }),
+                (CREATE, 0, {
+                    'product_id': part2.id,
+                    'product_tmpl_id': part2.product_tmpl_id.id,
+                }),
+            ],
+        })
+
+        ro_form = Form(self.env['repair.order'])
+        ro_form.product_id = repaired
+        ro = ro_form.save()
+        ro.action_validate()
+
+        with ro_form.move_ids.new() as ro_line:
+            ro_line.repair_line_type = 'add'
+            ro_line.product_id = kit
+        ro_form.save()
+
+        self.assertEqual(len(ro_form.move_ids), 2, "Repair order moves should correspond to the kit components")
```

Comment:
Using the form is a bit weird, it's better to use the record
Also, it could be nice to also ensure that `move_ids` are based on the components: either using `assertRecordValues` (and get rid of the `assertEqual`), or using another `assertEqual` on `ro.move_ids.product_id`. That way, you can also remove the error message as the assert will be clear enough